activerecord/lib/active_record/relation/batches.rb

```diff
@@ -435,20 +435,19 @@ def batch_on_unloaded_relation(relation:, start:, finish:, load:, cursor:, order
           if load
             records = batch_relation.records
             values = records.pluck(*cursor)
-            yielded_relation = where(cursor => values)
+            yielded_relation = batch_relation.where(cursor => values)
```

Comment:
Hum, this will re-add a lot of criterias, no?

I think all we really want is the ORDER clause, if present, right?

Reply:
Changed the implementation.

> if present, right?

In something like `Post.order(:created_at).in_batches` - that `order(:created_at)` is ignored. This has not changed, this is how it is currently on `main`.

What this PR fixes is that currently batching returns relations where only `IN (id1, ...idn)` set or where `id >= x and id < y` is set. There is no `ORDER` specified. So, that `IN` or `>= and <` conditions can return unordered records from the database. This can be unexpected, because the user expects results to be ordered by the batching column.